robot/packages/cloud_package_manager.go

```diff
@@ -353,11 +378,34 @@ func sanitizeURLForLogs(u string) string {
 	return parsed.String()
 }
 
-func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config.PackageConfig) error {
-	// TODO(): validate integrity of directory.
-	if dirExists(p.LocalDataDirectory(m.packagesDir)) {
-		m.logger.Debug("Package already downloaded, skipping.")
-		return nil
+// checkNonemptyPaths returns true if all required paths are present and non-empty.
+// This exists because we have no way to check the integrity of modules *after* they've been unpacked.
+// (We do look at checksums of downloaded tarballs, though). Once we have a better integrity check
+// system for unpacked modules, this should be removed.
+func checkNonemptyPaths(dataDir, packageName string, logger logging.Logger, paths []string) bool {
+	missingOrEmpty := 0
+	for _, nePath := range paths {
+		// note: os.Stat treats symlinks as their destination. os.Lstat would stat the link itself.
+		info, err := os.Stat(path.Join(dataDir, nePath))
```

Comment:
```suggestion
		info, err := os.Stat(path.Join(nePath))
```

nePath's are the module exe path which is an absolute path

Reply:
Ah oops
```suggestion
		info, err := os.Stat(nePath)
```

That

Reply:
my test didn't cover for this + should have

added a 'doesn't re-download when file isn't corrupt' case, made sure it failed with path.Join(dataDir, absPath)